staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -600,6 +778,36 @@ func (p *sharedProcessor) addListenerLocked(listener *processorListener) {
 	p.syncingListeners = append(p.syncingListeners, listener)
 }
 
+func (p *sharedProcessor) removeListener(index int) {
+	handler := p.listeners[index].handler
+	p.listeners = append(p.listeners[:index], p.listeners[index+1:]...)
+	for i := len(p.syncingListeners) - 1; i >= 0; i-- {
+		if p.syncingListeners[i].handler == handler {
+			p.syncingListeners = append(p.syncingListeners[:i], p.syncingListeners[i+1:]...)
+		}
+	}
+}
+
+func (p *sharedProcessor) removeEventHandler(handler ResourceEventHandler) bool {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	for i, listener := range p.listeners {
+		if listener.handler == handler {
```

Comment:
As we discussed before, this will panic if the handler is not comparable. Left it as is for now, but let me know if you think there’s an easy way to check if comparison is possible that we should be doing here instead of just letting the informer panic.

Reply:
add a comment here and above noting that